build_library/build_image_util.sh

```diff
@@ -737,9 +737,16 @@ EOF
   done
   sudo "${root_fs_dir}"/usr/sbin/flatcar-tmpfiles "${root_fs_dir}"
   # Now that we used the tmpfiles for creating /etc we delete them because
-  # the L, d, and C entries cause upcopies. Also filter out rules with ! or - but no other modifiers
+  # the L, d, D, and C entries cause upcopies. Also filter out rules with ! or - but no other modifiers
   # like + or = which explicitly recreate files.
-  sudo sed -i '/^[CLd]-*!*-*[ \t]*\/etc\//d' "${root_fs_dir}"/usr/lib/tmpfiles.d/*
+  # But before filtering, first store rules that would recreate missing files
+  # to /usr/share/flatcar/etc-no-whiteouts so that we can ensure that
+  # no overlayfs whiteouts exist for these files (example: /etc/resolv.conf).
+  # These rules are combined with the + modifier in addition.
+  # Other rules like w, e, x, do not create files that don't exist.
+  # Note: '-' must come first in the modifier pattern.
+  grep -Ph '^[fcCdDLvqQpb][-=~^!+]*[ \t]*/etc' "${root_fs_dir}"/usr/lib/tmpfiles.d/* | grep -oP '/etc[^ \t]*' | sudo_clobber "${root_fs_dir}"/usr/share/flatcar/etc-no-whiteouts
+  sudo sed -i '/^[CdDL][-=~^!]*[ \t]*\/etc\//d' "${root_fs_dir}"/usr/lib/tmpfiles.d/*
```

Comment:
Is this resolv.conf thing new in some way? I kicked off an image and grepping for any /etc entry in tmpfiles shows no results for resolv.conf:

```
core@localhost ~ $ grep VERSION= /etc/os-release 
VERSION=3802.0.0+weekly-package-updates-2
core@localhost ~ $ grep -Ph '^.[-=~^!+]*[ \t]*/etc' /usr/lib/tmpfiles.d/*
d   /etc                              -   -   -   -   -
Z   /etc/passwd                       -   -   -   -   -
Z   /etc/shadow                       -   -   -   -   -
Z   /etc/group                        -   -   -   -   -
Z   /etc/gshadow                      -   -   -   -   -
L+  /etc/ld.so.conf     -   -   -   -   ../usr/lib/ld.so.conf
L+ /etc/mtab - - - - ../proc/self/mounts
f       /etc/flatcar/update.conf       0644    -    -    -       -
f^ /etc/motd.d/50-provision.conf - - - - login.motd
f^ /etc/issue.d/50-provision.conf - - - - login.issue
f^ /etc/hosts - - - - network.hosts
f /etc/environment		0755 root root	-	-
r /etc/dbus-1/system.d/UpdateEngine.conf - - - 0 -
```